File: Golang/14.最长公共前缀.go
package leetcode

func longestCommonPrefix(strs []string) string {
	ans := strs[0]
	for i := 0; i < len(strs); i++ {
		for j := 0; j < len(ans); j++ {
			if j == len(strs[i]) || ans[j] != strs[i][j] {
				ans = ans[:j]
				break
			}
		}
	}
	return ans
}

File: TypeScript/14.最长公共前缀.ts
function longestCommonPrefix(strs: string[]): string {
    let ans: string = strs[0];
    for (let i = 0; i < strs.length; ++i) {
        for (let j = 0; j < ans.length; ++j) {
            if (j === strs[i].length || ans[j] !== strs[i][j]) {
                ans = ans.slice(0, j);
                break;
            }
        }
    }
    return ans;
};
